docs/content/docs/development/write-table.md

```diff
@@ -192,3 +192,95 @@ There are three main places in the Table Store's sink writer that take up memory
 - The memory consumed by compaction for reading files, it can be adjusted by the
   `num-sorted-run.compaction-trigger` option to change the maximum number of files to be merged.
 - The memory consumed by writing file, which is not adjustable.
+
+
+## Scale Bucket
+
+Since the LSM trees are built against each bucket, the number of total buckets dramatically influences the performance.
+Table Store allows users to tune bucket numbers by `ALTER TABLE` command and reorganize data layout by `INSERT OVERWRITE` 
+without recreating the table/partition. When executing overwrite jobs, the framework will automatically scan the data with
+the bucket number recorded in manifest file and hash the record according to the current bucket numbers.
+
+#### Rescale Overwrite
+```sql
+-- scale number of total buckets
+ALTER TABLE table_dentifier SET ('bucket' = '...')
+
+-- reorganize data layout of table/partition
+INSERT OVERWRITE table_identifier [PARTITION (part_spec)]
+SELECT ... 
+FROM table_identifier
+[WHERE part_spec]
+``` 
+
+Please beware that
+- `ALTER TABLE` only modifies the table's metadata and will **NOT** reorganize or reformat existing data. 
+  Reorganize exiting data must be achieved by `INSERT OVERWRITE`.
+- Scale bucket number does not influence the read and running write jobs.
+- Once the bucket number is changed, any new `INSERT INTO` jobs without reorganize table/partition 
+  will throw a `TableException` with message like 
+  ```text
+  Try to write table/partition ... with a new bucket num ..., 
+  but the previous bucket num is ... Please switch to batch mode, 
+  and perform INSERT OVERWRITE to rescale current data layout first.
+  ```
+
+
+{{< hint info >}}
+__Note:__ Currently, scale bucket is only supported for tables without enabling log system.
```

Comment:
I thought about it, currently the kafka topic is managed by the user, we seem to only need to remind the user to modify the number of kafka topic partitions can be.